### PR TITLE
[CELEBORN-870][DOC] Document on usage together with Gluten

### PR DESCRIPTION
**Pull request by kerwin-zk**
### What changes were proposed in this pull request?
As title.

### Why are the changes needed?
As title.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Manual test.

### CODE REVIEW

File: docs/developers/glutensupport.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
```
Comment (pan3793) [inline docs/developers/glutensupport.md:9]: nit: fix the intend

File: docs/developers/glutensupport.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+---
+
+# Gluten Support
+## Velox Backend
+
+[Gluten](https://github.com/oap-project/gluten) with velox backend supports Celeborn as remote shuffle service. Below introduction is used to enable this feature
+
+First refer to this URL(https://github.com/oap-project/gluten/blob/main/docs/get-started/Velox.md) to build Gluten with velox backend.
+
+When compiling the Gluten java module, you need to add rss, as follows:
```
Comment (pan3793) [inline docs/developers/glutensupport.md:24]:
```suggestion
When compiling the Gluten Java module, it's required to enable `rss` profile, as follows:
```

File: docs/developers/glutensupport.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+```
+mvn clean package -Pbackends-velox -Pspark-3.3 -Prss -DskipTests
+```
+
+Then add the paths of Gluten and Spark Celeborn Client packages to class path.
+
+Currently to use Gluten following configurations are required in spark-defaults.conf
```
Comment (pan3793) [inline docs/developers/glutensupport.md:32]:
```suggestion
Currently to use Gluten following configurations are required in `spark-defaults.conf`
```

File: docs/developers/glutensupport.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+First refer to this URL(https://github.com/oap-project/gluten/blob/main/docs/get-started/Velox.md) to build Gluten with velox backend.
+
+When compiling the Gluten java module, you need to add rss, as follows:
+
+```
+mvn clean package -Pbackends-velox -Pspark-3.3 -Prss -DskipTests
+```
+
+Then add the paths of Gluten and Spark Celeborn Client packages to class path.
```
Comment (pan3793) [inline docs/developers/glutensupport.md:30]:
```suggestion
Then add the Gluten and Spark Celeborn Client packages to your Spark application's classpath(usually add them into `$SPARK_HOME/jars`).

- Celeborn: xxxx.jar
- Gluten: xxx.jar
```

**Comment by pan3793** [comment]
it's better to document the version matrix too.

File: docs/developers/glutensupport.md (hunk 4 above)
Comment (pan3793) [reply docs/developers/glutensupport.md]: replace the `xxxx.jar` with real pattern ...

Comment (kerwin-zk) [reply docs/developers/glutensupport.md]: I think this part can be removed, the package name is not unique, there are different versions.

File: docs/developers/glutensupport.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+When compiling the Gluten Java module, it's required to enable `rss` profile, as follows:
+
+```
+mvn clean package -Pbackends-velox -Pspark-3.3 -Prss -DskipTests
+```
+
+Then add the Gluten and Spark Celeborn Client packages to your Spark application's classpath(usually add them into `$SPARK_HOME/jars`).
+
+- Celeborn: celeborn-client-spark-3-shaded_2.12-0.3.0-incubating.jar
```
Comment (zwangsheng) [inline docs/developers/glutensupport.md:34]: There's a little doubt about that, we should use gluten compiled for `gluten-celeborn-XXX.jar` or `Celeborn-client-spark-3-shaded_2.12-XXX-incubating.jar`?

Comment (kerwin-zk) [reply docs/developers/glutensupport.md:34]: The previous Gluten+Celeborn integrated ```celeborn-client-spark-3-shaded_2.12-XXX-incubating.jar``` into the Gluten compilation package, so users only need the Gluten package before.  After this pr (https://github.com/oap-project/gluten/pull/2530) was merged, the Celeborn package was removed from Gluten, so users need to put ```celeborn-client-spark-3-shaded_2.12-XXX-incubating.jar```  in the class path or add it into `$SPARK_HOME/jars`.

**Review by pan3793 — APPROVED** [review]
LGTM, please resolve conflicts

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merged to main.

**Pull request closed without merging**